### PR TITLE
Retry Cockroach transactions

### PR DESCRIPTION
**Pull request by tirsen**

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt [hunk 1]
```diff
@@ -323,7 +323,7 @@ internal class RealTransacter private constructor(
   }
 
   private fun isMessageRetryable(th: SQLException) =
-      isConnectionClosed(th) || isVitessTransactionNotFound(th)
+      isConnectionClosed(th) || isVitessTransactionNotFound(th) || isCockroachRestartTransaction(th)
```
Comment (mightyguava) [inline misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: I'm curious why this hasn't been needed for TiDB

Comment (tirsen) [reply misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: I bet we gotta do something similar but maybe it doesn't fail as often?

Comment (mightyguava) [reply misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: TiDB does some internal retries for this kind of conflict. I'm curious if cockroach expects the user to retry.

Comment (tirsen) [reply misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: Oh right it was those retries that Jepsen commented on. I thought they were disabled by default?

Comment (mightyguava) [reply misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt]: Uh, true, maybe they don't do that anymore

**Pull request merged**